frontend/talentsearch/src/js/components/aboutMeForm/AboutMeForm.tsx

```diff
@@ -0,0 +1,342 @@
+                description:
+                  "Placeholder displayed on the About Me form telephone field.",
+              })}
+              rules={{
+                required: intl.formatMessage(errorMessages.required),
+                pattern: {
+                  value: phoneNumberRegex,
+                  message: intl.formatMessage(errorMessages.telephone),
+                },
+              }}
+            />
+          </div>
+        </div>
+        <h2 data-h2-font-size="b(h3)" data-h2-font-weight="b(700)">
+          {intl.formatMessage({
+            defaultMessage: "Account Details",
+            description:
+              "Title for Account Details section of the About Me form",
+          })}
+        </h2>
+        <p>
+          {intl.formatMessage({
+            defaultMessage:
+              "This information is used to manage your account and to communicate.",
+            description:
+              "Description for Account Details section of the About Me form",
+          })}
+        </p>
+        <div data-h2-flex-item="b(1of1)" data-h2-padding="b(top, m)">
+          <div data-h2-padding="b(right, l)">
+            <Input
+              id="firstName"
+              name="firstName"
+              type="text"
+              label={intl.formatMessage({
+                defaultMessage: "First Name and middle Name",
+                description:
+                  "Label for first and middle name field in About Me form",
+              })}
+              rules={{
+                required: intl.formatMessage(errorMessages.required),
+              }}
+            />
+            <Input
+              id="lastName"
+              name="lastName"
+              type="text"
+              label={intl.formatMessage({
+                defaultMessage: "Last Name(s)",
+                description: "Label for last name field in About Me form",
+              })}
+              rules={{
+                required: intl.formatMessage(errorMessages.required),
+              }}
+            />
+            <Input
+              id="email"
+              name="email"
+              type="email"
+              label={intl.formatMessage({
+                defaultMessage: "Email",
+                description: "Label for email field in About Me form",
+              })}
+              rules={{
+                required: intl.formatMessage(errorMessages.required),
+              }}
+            />
+          </div>
+        </div>
+        <ProfileFormFooter mode="saveButton" />
+      </BasicForm>
+    </ProfileFormWrapper>
+  );
+};
+
+const AboutMeFormContainer: React.FunctionComponent = () => {
+  const intl = useIntl();
+
+  const [result] = useGetAboutMeQuery();
+  const { data, fetching, error } = result;
+
+  const [_, executeMutation] = useUpdateUserAsUserMutation();
+
+  const handleUpdateUser = (id: string, values: UpdateUserAsUserInput) => {
+    return executeMutation({ id, user: values }).then((res) => {
```

Comment:
Eslint would really like you to add a type annotation for `res` :)

Reply:
I relied on the TS Server type hinting to resolve this so I'm not entirely certain on it. However, it still compiles and tests succeed.

Resolved with [annotate update user mutation response](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2413/commits/8e616053e855b00a5927de70a17aa8a90771b908)